app/controllers/api/auth/registrations_controller.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class Api::Auth::RegistrationsController < DeviseTokenAuth::RegistrationsController
+  def create
+    invitation_token = request.headers[:InvitationToken]
+    return render_create_error_token_invalid if invitation_token.present? && !Team.invitation_token_valid?(invitation_token)
+    super do |resource|
+      if invitation_token.present?
+        team = Team.find_by(invitation_token: invitation_token)
```

Comment:
devise-token-auth が `ActiveRecord::RecordNotFound` も良きに計らってくれるのであれば `find_by!` しても良いかもですね。そうすると L6 のガードが必要なくなりそう。

```suggestion
        team = Team.find_by!(invitation_token: invitation_token)
```